appsdk/sdk.go

```diff
@@ -438,15 +437,16 @@ func (sdk *AppFunctionsSDK) Initialize() error {
 	sdk.appWg.Add(1)
 	go telemetry.StartCpuUsageAverage(sdk.appWg, sdk.appCtx, sdk.LoggingClient)
 
-	sdk.webserver = webserver.NewWebServer(&sdk.config, sdk.LoggingClient, mux.NewRouter())
+	sdk.webserver = webserver.NewWebServer(&sdk.config, sdk.secretProvider, sdk.LoggingClient, mux.NewRouter())
```

Comment:
The sdk is added to the request's context when adding route, so we shouldn't need to add these here.
The context can be accessed by request.Context(). The sdk is added with the id sdk.SDKKey.

sdk.secretProvider will need to be exported, i.e. sdk.SecretProvider

Reply:
Oh good catch, I was thinking of ways to remove that reference.